igor-monitor-jenkins/src/main/java/com/netflix/spinnaker/igor/jenkins/JenkinsBuildMonitor.java

```diff
@@ -279,16 +283,52 @@ private void postEvent(final Project project, final String master) {
       return;
     }
 
+    JenkinsBuildEvent event = new JenkinsBuildEvent(new JenkinsBuildContent(project, master));
     AuthenticatedRequest.allowAnonymous(
         () -> {
           echoService.ifPresent(
-              echo ->
-                  echo.postEvent(new JenkinsBuildEvent(new JenkinsBuildContent(project, master))));
+              echo -> {
+                try {
+                  echo.postEvent(event);
+                } catch (RetrofitError e) {
+                  String rawResponse = "failed to retrieve response body";
+                  try {
+                    rawResponse = getResponseBody(e);
+                  } catch (IOException ioe) {
+                    log.error(
+                        "Failed to read response while handling error communicating with Echo",
+                        ioe);
+                  }
+                  log.error(
+                      "Failed to post jenkins build event to Echo ({}:{}): {}",
+                      master,
+                      project.getName(),
+                      rawResponse,
+                      e);
+                  throw e;
+                }
+              });
```

Comment:
The exception being thrown from Retrofit was totally unhelpful, so I wrote some more error handling logic to figure out what was going on. Figure why not leave it in.